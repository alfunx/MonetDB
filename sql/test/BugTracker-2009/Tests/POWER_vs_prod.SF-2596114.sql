-- disable parallelism (mitosis & dataflow) to avoid ambiguous results 
set optimizer='sequential_pipe';
create table SF_2596114 (x double);
explain select power(x,2) from SF_2596114;
explain select x*x from SF_2596114;
drop table SF_2596114;
