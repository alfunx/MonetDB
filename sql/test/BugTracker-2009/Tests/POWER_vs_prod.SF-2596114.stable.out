stdout of test 'POWER_vs_prod.SF-2596114` in directory 'sql/test/BugTracker-2009` itself:


# 23:02:29 >  
# 23:02:29 >   mserver5 "--config=/ufs/manegold/_/scratch0/Monet/Testing/Stable/prefix.--enable-strict_--disable-optimize_--enable-debug_--enable-assert/etc/monetdb5.conf" --debug=10 --set gdk_nr_threads=0 --set "monet_mod_path=/ufs/manegold/_/scratch0/Monet/Testing/Stable/prefix.--enable-strict_--disable-optimize_--enable-debug_--enable-assert/lib64/MonetDB5:/ufs/manegold/_/scratch0/Monet/Testing/Stable/prefix.--enable-strict_--disable-optimize_--enable-debug_--enable-assert/lib64/MonetDB5/lib:/ufs/manegold/_/scratch0/Monet/Testing/Stable/prefix.--enable-strict_--disable-optimize_--enable-debug_--enable-assert/lib64/MonetDB5/bin" --set "gdk_dbfarm=/ufs/manegold/_/scratch0/Monet/Testing/Stable/prefix.--enable-strict_--disable-optimize_--enable-debug_--enable-assert/var/MonetDB5/dbfarm" --set "sql_logdir=/ufs/manegold/_/scratch0/Monet/Testing/Stable/prefix.--enable-strict_--disable-optimize_--enable-debug_--enable-assert/var/MonetDB5/sql_logs"   --set mapi_open=true --set xrpc_open=true --set mapi_port=33742 --set xrpc_port=44239 --set monet_prompt= --trace  "--dbname=mTests_src_test_BugTracker-2009" --set mal_listing=0 "--dbinit= include sql;" ; echo ; echo Over..
# 23:02:29 >  

# MonetDB server v5.10.3, based on kernel v1.28.3
# Serving database 'mTests_src_test_BugTracker-2009', using 4 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs dynamically linked
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2009 MonetDB B.V., all rights reserved
# Visit http://monetdb.cwi.nl/ for further information
# Listening for connection requests on mapi:monetdb://rig.ins.cwi.nl:33742/
# MonetDB/SQL module v2.28.1 loaded

Ready.
#function user.main():void;
#    clients.quit();
#end main;


# 15:46:08 >  
# 15:46:08 >  mclient -lsql -i -umonetdb -Pmonetdb --host=rig --port=35464 
# 15:46:08 >  

% .explain # table_name
% mal # name
% clob # type
% 100 # length
function user.s3_1(A0:dbl):void;
    X_35:void := querylog.define("explain select power(x,2) from sf_2596114;","sequential_pipe",22);
    X_18 := bat.new(nil:oid,nil:str);
    X_26 := bat.append(X_18,"sys.L");
    X_21 := bat.new(nil:oid,nil:str);
    X_28 := bat.append(X_21,"power_x");
    X_22 := bat.new(nil:oid,nil:str);
    X_30 := bat.append(X_22,"double");
    X_23 := bat.new(nil:oid,nil:int);
    X_32 := bat.append(X_23,53);
    X_25 := bat.new(nil:oid,nil:int);
    X_34 := bat.append(X_25,0);
    X_2 := sql.mvc();
    C_3:bat[:oid] := sql.tid(X_2,"sys","sf_2596114");
    X_6:bat[:dbl] := sql.bind(X_2,"sys","sf_2596114","x",0);
    (C_9,r1_9) := sql.bind(X_2,"sys","sf_2596114","x",2);
    X_12:bat[:dbl] := sql.bind(X_2,"sys","sf_2596114","x",1);
    X_14 := sql.delta(X_6,C_9,r1_9,X_12);
    X_15 := algebra.projection(C_3,X_14);
    X_16:bat[:dbl] := batmmath.pow(X_15,A0);
    sql.resultSet(X_26,X_28,X_30,X_32,X_34,X_16);
end user.s3_1;
#explain select x*x from SF_2596114;
% .explain # table_name
% mal # name
% clob # type
% 93 # length
function user.s4_1():void;
    X_34:void := querylog.define("explain select x*x from sf_2596114;","sequential_pipe",22);
    X_17 := bat.new(nil:oid,nil:str);
    X_25 := bat.append(X_17,"sys.L");
    X_20 := bat.new(nil:oid,nil:str);
    X_27 := bat.append(X_20,"sql_mul_x");
    X_21 := bat.new(nil:oid,nil:str);
    X_29 := bat.append(X_21,"double");
    X_22 := bat.new(nil:oid,nil:int);
    X_31 := bat.append(X_22,53);
    X_24 := bat.new(nil:oid,nil:int);
    X_33 := bat.append(X_24,0);
    X_1 := sql.mvc();
    C_2:bat[:oid] := sql.tid(X_1,"sys","sf_2596114");
    X_5:bat[:dbl] := sql.bind(X_1,"sys","sf_2596114","x",0);
    (C_8,r1_8) := sql.bind(X_1,"sys","sf_2596114","x",2);
    X_11:bat[:dbl] := sql.bind(X_1,"sys","sf_2596114","x",1);
    X_13 := sql.delta(X_5,C_8,r1_8,X_11);
    X_14 := algebra.projection(C_2,X_13);
    X_15:bat[:dbl] := batcalc.*(X_14,X_14);
    sql.resultSet(X_25,X_27,X_29,X_31,X_33,X_15);
end user.s4_1;
#drop table SF_2596114;

# 23:02:30 >  
# 23:02:30 >  Done.
# 23:02:30 >  

